### PR TITLE
LTM-89 - Faulty GPT evaluations

### PR DESCRIPTION
**Pull request by dcasbol**
Use code evaluations in _Trigger Response_ and _Sally Anne_ tests, instead of GPT.
Improved delivery of messages to improve comprehension and avoid confusing the agent.
Refactor of evaluation methods and base class.
Improved reasoning strings.

### CODE REVIEW

File: data/gpt_generation_prompts/2-1_prospective_memory_test.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "content":"Formulate an entirely novel fictitious quote from a fictitious person in single quotes ''. Say that it is a quote from that person",
-  "question":"Instruct the subject to recite the above quote by appending it to the response after the next `n` statements by the user. Where `n` is an integer between 1 and 10.",
+  "question":"Instruct the subject to recite the above quote by appending it to the response of the `n`th new message. Where `n` is an integer between 1 and 10.",
```
Comment (JosephDavidsonKSWH) [inline data/gpt_generation_prompts/2-1_prospective_memory_test.json:3]: This results in some wonky instructions:
```
Recite the quote from Captain Emilia Starfinder in your 1th new message.
Please recite the quote from Alistair Fenwick in your 7th new message.
To complete this task, recite the quote from Dr. Elara Moresby in the response of the 7th new message.
```

Might be clearer to tell the generator to generate an integer, then ask that the agent "append the quote to its response to to the nth message after this one."

Comment (dcasbol) [reply data/gpt_generation_prompts/2-1_prospective_memory_test.json:3]: Right. I'll try that.

Comment (dcasbol) [reply data/gpt_generation_prompts/2-1_prospective_memory_test.json]: That seems to make it much clearer. I've also added some changes to use numbers between 2 and 10 (avoiding 1, which might be extra confusing), and to display them as `2nd`, `3rd`, etc.

Comment (JosephDavidsonKSWH) [reply data/gpt_generation_prompts/2-1_prospective_memory_test.json]: Yeah the generation looks much better!

**Review by JosephDavidsonKSWH — APPROVED** [review]

**Pull request merged**